aerogear/aerogear-js-sdk
# feat: swapped out custom interface for Localforage

### Description
This PR adds the ability to use async storage as default in the sdk. 

It also splits out the offline mutation store into individual mutations. This reduces the need to stringify all of these mutations each time one is added or read.

Verification:

1. Use the showcase application and default config.
1. Clone this branch, build it and npm link it to the showcase application by: `npm link @aerogear/voyager-client`
1. Run the showcase application with `npm run start`
1. Create some items and everything should work fine
1. Go offline
1. Create some more items
1. Check async storage in your browser and you should see multiple entries depending on how many you created
1. Go online
1. Items should be replayed and synced to the server
1. Go offline
1. Create some items
1. Go online and restart the application
1. The previous offline items should be replayed and synced to the server
